gemini/evaluation/evaluating_prompts_at_scale_with_gemini_batch_prediction_api.ipynb

```diff
@@ -0,0 +1,1011 @@
+    "            \"bigqueryDestination\": {\"outputUri\": f\"bq://{text_generation_table}\"},\n",
+    "        },\n",
+    "    }\n",
+    "    headers = {\n",
+    "        \"Content-Type\": \"application/json\",\n",
+    "        \"Authorization\": f\"Bearer {token}\",\n",
+    "    }\n",
+    "\n",
+    "    batch_response = requests.post(gemini_batch_url, json=request_data, headers=headers)\n",
+    "    print(batch_response.text)\n",
+    "\n",
+    "    # Get job id of the request.\n",
+    "    batch_response_json = batch_response.json()\n",
+    "    job_name = batch_response_json[\"name\"]\n",
+    "    job_id = job_name.split(\"/\")[-1]\n",
+    "\n",
+    "    # Get state of the request.\n",
+    "    job_state = batch_response_json[\"state\"]\n",
+    "    return job_id, job_state\n",
+    "\n",
+    "\n",
+    "job_id, job_state = create_batch_prediction_job(\n",
+    "    project_id=PROJECT_ID,\n",
+    "    model_id=MODEL_ID,\n",
+    "    prompts_table=PROMPTS_TABLE,\n",
+    "    text_generation_table=text_generation_table,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "dLlsbnbXPpgK"
+   },
+   "source": [
+    "### Check state of the Gemini Batch Prediction request until completed"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "B9atOKsOUCs0",
+    "outputId": "690f090f-7f89-4a26-9fd1-b2d58998cc0f"
+   },
+   "outputs": [],
+   "source": [
+    "def check_job_state(project_id, job_id, last_job_state):\n",
+    "    \"\"\"Checks status of the Gemini batch text generation request.\"\"\"\n",
+    "\n",
+    "    get_status_url = (\n",
+    "        f\"https://us-central1-aiplatform.googleapis.com/v1/projects/{project_id}/\"\n",
+    "        f\"locations/us-central1/batchPredictionJobs/{job_id}\"\n",
+    "    )\n",
+    "\n",
+    "    # Get Authentication token.\n",
+    "    creds, _ = google.auth.default()\n",
+    "    auth_req = google.auth.transport.requests.Request()\n",
+    "    creds.refresh(auth_req)\n",
+    "    token = creds.token\n",
+    "\n",
+    "    headers = {\n",
+    "        \"Content-Type\": \"application/json\",\n",
+    "        \"Authorization\": f\"Bearer {token}\",\n",
+    "    }\n",
+    "\n",
+    "    job_state = last_job_state\n",
+    "    now = datetime.datetime.now()\n",
+    "    print(f\"Status {now}: {job_state}\")\n",
+    "\n",
+    "    while job_state not in STOP_STATES:\n",
+    "        state_response = requests.get(get_status_url, headers=headers)\n",
+    "        state_response_json = state_response.json()\n",
+    "        job_state = state_response_json[\"state\"]\n",
+    "        now = datetime.datetime.now()\n",
+    "        print(f\"Status {now}: {job_state}\")\n",
+    "        if job_state not in STOP_STATES:\n",
+    "            time.sleep(30)\n",
+    "\n",
+    "\n",
```

Comment:
Same as above, please use the SDK. We shouldn't encourage users to do manual REST API calls.